### PR TITLE
doc(assert): assert action documentation

### PR DESCRIPTION
**Pull request by KarimGl**

### CODE REVIEW

File: docs/actions/assert.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+### Example
+=== "Kotlin"
+    ``` kotlin
+    JsonAssertTask(
+        documentVariable = "{" +
+                           "    \"id\": 1," +
+                           "    \"name\": \"mouse\"," +
+                           "    \"price\": 12" +
+                           "}",
+        expectationsVariable = "{" +
```
Comment (nbrouand) [inline docs/actions/assert.md:103]: wrong inputs on its json validation not json assert

File: docs/actions/assert.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+## Json compare
+compare two json inputs(the hole content or only some nodes) using a comparison mode.
+
+=== "Inputs"
+
+    | Required | Name            | Type        |     Default      |    Description                  |
+    |:--------:|:----------------|:------------|:----------------:|:-------------------------------:|
+    |    *     | `document1`     | String      |                  | first json                      |
+    |    *     | `document2`     | String      |                  | second json                     |
+    |    *     | `comparingPaths`| Map         | Map.of("$", "$") | to be comapared nodes [json paths](https://github.com/json-path/JsonPath){:target="_blank"} |
```
Comment (nbrouand) [inline docs/actions/assert.md:210]: comapared

File: docs/actions/assert.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+=== "Outputs"
+    No outputs
+
+
+### Example
+=== "Kotlin"
+    ``` kotlin
+    JsonCompareTask(
+        document1 = "{\"string\": \"string_value\", \"object\":{\"number\":3}, \"list\": [1, 2, 3]}",
```
Comment (nbrouand) [inline docs/actions/assert.md:221]: make the example with all inputs

File: docs/actions/assert.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    )
+    ```
+!!! note "mode can be"
+
+    | mode       |          desscription                                                                                    |
+    |:----------:|:---------------------------------------------------------------------------------------------------------|
+    | `STRICT`   | strict comparison                                                                                        |
+    | `LENIENT`  | even if the document1 contains extended fields (which document2 doesn't have), the test will still pass  |
+
+# Step validation
```
Comment (nbrouand) [inline docs/actions/assert.md:232]: I will precise this at the begining at the page

**Review by nbrouand — CHANGES_REQUESTED** [review]

File: docs/actions/assert.md [hunk 5]
```diff
@@ -0,0 +1,235 @@
+# Assertions
```
Comment (bessonm) [inline docs/actions/assert.md:1]: Remove head title, there is a default one taken from the menu entry.

File: docs/actions/assert.md [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+## Assert
+assert that all evaluated inputs are true.
```
[the remark this answers is not in the document]
Comment (bessonm) [reply docs/actions/assert.md:4]: Dry sentences are a bit harsh to understand. I would prefer longer and clearer sentences.
ex. "This action takes a list of assertions written using SpEL and validates they are all true."

File: docs/actions/assert.md [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+=== "Inputs"
+
+    | Required | Name            | Type        |     Default     |
+    |:--------:|:----------------|:------------|:---------------:|
+    |    *     | `asserts`       | List of Map |                 |
+
+=== "Outputs"
+    No outputs
```
Comment (bessonm) [inline docs/actions/assert.md:12]: We should decide, do we put this in the output tab or outside the tab ?
ex. https://www.chutney-testing.com/actions/jms/#jms-sender & https://www.chutney-testing.com/actions/ssh/#download

File: docs/actions/assert.md [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+    ``` kotlin
+    CompareTask(
+        actual = "chutney",
+        expected = "chutney",
+        mode = "EQUALS" // case insensitive
+    ) 
+    ```
+!!! note "mode can be"
+
+    | mode                             |          desscription                  |
```
Comment (bessonm) [inline docs/actions/assert.md:192]: typo _desscription_

File: docs/actions/assert.md [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+    |:--------------------------------:|:---------------------------------------|
+    | `equals`                         | actual and expected are equals         |
+    | `not equals` / `not-equals`      | actual and expected are not equals     |
+    | `contains`                       | actual contains expectd                |
+    | `not-contains` / `not contains`  | actual doesn't contain expectd         |
+    | `greater-than` / `greater than`  | actual is greater than expected        |
```
Comment (bessonm) [inline docs/actions/assert.md:198]: typo _expectd_

File: docs/actions/assert.md [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+    | `less-than` / `less than`        | actual is less than expected           |
+
+## Json compare
+compare two json inputs(the hole content or only some nodes) using a comparison mode.
```
Comment (bessonm) [inline docs/actions/assert.md:202]: ahah, beware ! don't mistake the whole with a hole :)

File: docs/actions/assert.md [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+### Example
+=== "Kotlin"
+    ``` kotlin
+    CompareTask(
+        actual = "chutney",
+        expected = "chutney",
+        mode = "EQUALS" // case insensitive
+    ) 
+    ```
+!!! note "mode can be"
```
Comment (bessonm) [inline docs/actions/assert.md:190]: "available modes"

File: docs/actions/assert.md (hunk 4 above)
Comment (bessonm) [reply docs/actions/assert.md:232]: Same, at the beginning and I would give a lot more explanations. 
if you don't know how to do right now, no worry, we can improve it later

File: docs/actions/assert.md (hunk 5 above)
Comment (KarimGl) [reply docs/actions/assert.md]: this is a subtitle. The page contains two subtitles: assertions and validations

File: docs/actions/assert.md (hunk 7 above)
Comment (KarimGl) [reply docs/actions/assert.md]: I prefer to represent outputs like inputs using tabs

**Review by bessonm — APPROVED** [review]

**Review by boddissattva — APPROVED** [review]

**Review by nbrouand — APPROVED** [review]

**Pull request merged**